### PR TITLE
Fix omniauth oauth2 dependency

### PR DESCRIPTION
**Pull request by franciscoj**
### :tophat: What?
- Fix omniauth-oauth2 dependency to 1.3.1 since it was breaking the integration (check here https://github.com/intridea/omniauth-oauth2/issues/81 and here https://github.com/intridea/omniauth-oauth2/issues/28 for more info)
- Stop dealing with SCRIPT name.
- Show some love to the specs
- Add a changelog.

### CODE REVIEW

**Comment by csaura** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Change Log
+
+## [Unreleased]
+### Added
+- Test on ruby 2.1 and 2.2 too.
+### Changed
+- cleaned up the code a bit.
```
Comment (jrom) [inline CHANGELOG.md:11]: Probably this doesn't belong to a CHANGELOG...

**Comment by jhbabon** [comment]
:+1:

File: lib/omniauth/strategies/redbooth.rb [hunk 2]
```diff
@@ -2,62 +2,34 @@
...
-      # the `path_prefix` cannot start with a slash. Because of that
-      # `request.url` becomes host:portpath (notice the lack of a slash).
-      #
-      # This code sets the `SCRIPT_NAME` environment variable to a slash
-      # if it's empty (meaning that it's the root) and then calls
-      # the parent code which instantiates a request.
-      def request
-        @env['SCRIPT_NAME'] = '/' if @env['SCRIPT_NAME'] == ''
-        super
+        @raw_info ||=
```
Comment (jrom) [inline lib/omniauth/strategies/redbooth.rb:31]: The whole `SCRIPT_NAME` thing was supposed to help with engines mounted on a path that wasn't root. Did you test this scenario with the changed code?

**Comment by jrom** [comment]
:mailbox_with_mail: :+1:

File: lib/omniauth/strategies/redbooth.rb (hunk 2 above)
Comment (franciscoj) [inline lib/omniauth/strategies/redbooth.rb:31]: Yes, we were using the engines and the configuration in a wrong way. They should be working now. In any case if we need this patch somewhere in one of our integrations I would rather do it there than add a fix in the gem which can potentially break all other 3rd party integrations.

File: lib/omniauth/strategies/redbooth.rb [hunk 3]
```diff
@@ -2,62 +2,34 @@
 
 module OmniAuth
   module Strategies
+    # @see https://redbooth.com/api/api-docs/
     class Redbooth < OmniAuth::Strategies::OAuth2
-      # Give your strategy a name.
-      option :name, 'redbooth'
+      REDBOOTH_URL = 'https://redbooth.com'
```
Comment (sauloperez) [inline lib/omniauth/strategies/redbooth.rb]: freeze it, please

File: lib/omniauth/strategies/redbooth.rb [hunk 4]
```diff
@@ -2,62 +2,34 @@
...
 
+      option :name, 'redbooth'
```
Comment (sauloperez) [inline lib/omniauth/strategies/redbooth.rb:9]: this one too

File: spec/omniauth/strategies/redbooth_spec.rb [hunk 5]
```diff
@@ -1,102 +1,118 @@
 require 'spec_helper'
 
-describe OmniAuth::Strategies::Redbooth do
+RSpec.describe OmniAuth::Strategies::Redbooth do
```
Comment (sauloperez) [inline spec/omniauth/strategies/redbooth_spec.rb:3]: Is `RSpec` really needed?

**Comment by sauloperez** [comment]
:postal_horn: :+1:

File: spec/omniauth/strategies/redbooth_spec.rb (hunk 5 above)
Comment (franciscoj) [inline spec/omniauth/strategies/redbooth_spec.rb:3]: I really prefer to not deal with the global namespace. Latest versions of `rspec` support doing this and in addition they encourage it. So I consider it a good practice.

Comment (jhbabon) [inline spec/omniauth/strategies/redbooth_spec.rb:3]: In the past `RSpec` added this methods `describe` to the global namespace, `Kernel` (or maybe `Object`?), but that is not really cool. In the latest versions of `RSpec` you can remove this feature and use `RSpec` to open the spec definitions, as you can see in this example.

I would say is better to keep it, is more explicit.

File: lib/omniauth/strategies/redbooth.rb (hunk 4 above)
Comment (franciscoj) [inline lib/omniauth/strategies/redbooth.rb:9]: I don't know where or how omniauth is going to use it, I'd rather not freeze something that might be used somewhere in a way which breaks something.

**Pull request merged**